snapcraft/internal/errors.py

```diff
@@ -295,6 +295,17 @@ def __init__(self, message):
         super().__init__(message=message)
 
 
+class ToolMissingError(SnapcraftError):
+
+    fmt = (
+        "A tool snapcraft depends on could not be found: {command_name!r}.\n"
+        "Ensure the tool is installed and available, and try again."
```

Comment:
This is a workaround, but in reality, for all the use-cases of this function so far (e.g. xdelta3, mksquashfs, etc.) all of these tools should ALWAYS be found, and if they're not, it's a bug we should know about. Worth adding "please log a bug" to this?